src/wp-includes/media.php

```diff
@@ -1934,6 +1942,45 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 	return $image;
 }
 
+/**
+ * Add `decoding` attribute to an `img` HTML tag.
+ *
+ * The `decoding` attribute allows developers to indicate whether the
+ * browser can decode the image off the main thread (`async`), on the
+ * main thread (`sync`) or as determined by the browser (`auto`).
+ *
+ * By default WordPress adds `decoding="async"` to images but developers
+ * can use the {@see 'wp_img_tag_add_decoding_attr'} filter to modify this
+ * to remove the attribute or set it to another accepted value.
+ *
+ * @since 6.0.0
+ *
+ * @param string $image   The HTML `img` tag where the attribute should be added.
+ * @param string $context Additional context to pass to the filters.
+ * @return string Converted `img` tag with `decoding` attribute added.
+ */
+function wp_img_tag_add_decoding_async_attr( $image, $context ) {
```

Comment:
I'd suggest we call this function `wp_img_tag_add_decoding_attr`, both for consistency with the filter name (which doesn't include `async`) as well as for consistency with the other similar functions (which only include the attribute name, but not the value name we choose by default). I also think that makes more sense semantically, since the filter here may actually set another value than `async`. So really this function just adds a `decoding` attribute, to one of the possible values.

Reply:
👍🏼 Thanks for the naming tips! Changed in [`54511cf` (#2409)](https://github.com/WordPress/wordpress-develop/pull/2409/commits/54511cfe2b756f458abe4e079705305090400f32)